app/src/main/java/tech/ula/ui/AppListAdapter.kt

```diff
@@ -47,92 +65,177 @@ class AppListAdapter(
                 val activeAppSessions = activeSessions.filter { it.name == app.name }
                 val appIsActive = activeAppSessions.isNotEmpty()
                 if (appIsActive) {
-                    view?.setBackgroundResource(R.color.colorAccent)
+                    viewHolder.itemView.setBackgroundResource(R.color.colorAccent)
                 } else {
-                    view?.setBackgroundResource(R.color.colorPrimaryDark)
+                    viewHolder.itemView.setBackgroundResource(R.color.colorPrimaryDark)
                 }
 
                 val localFileLocator = LocalFileLocator(activity.filesDir.path, activity.resources)
                 viewHolder.imageView?.setImageURI(localFileLocator.findIconUri(app.name))
                 viewHolder.appName?.text = app.name.capitalize()
+                setAnimation(viewHolder.itemView, position)
             }
         }
+    }
+
+    private fun setAnimation(viewToAnimate: View, position: Int) {
+        if (position > lastPosition) {
+            val animationDelay = 200L
+            val animation = AnimationUtils.loadAnimation(activity, R.anim.item_animation_from_right)
+            viewToAnimate.startAnimation(animation)
+            animation.startOffset = position * animationDelay
+            lastPosition = position
+        }
+    }
 
-        return view as View
+    private fun bindOnClick(viewHolder: ViewHolder, selectedListItem: AppsListItem, onAppsItemClicked: OnAppsItemClicked, position: Int) {
+        viewHolder.itemView.setOnClickListener {
+            onAppsItemClicked.onAppsItemClicked(selectedListItem)
+        }
     }
 
-    override fun getItem(position: Int): AppsListItem {
-        return appsAndSeparators[position]
+    private fun bindOnCreateContextMenu(viewHolder: ViewHolder, onAppsCreateContextMenu: OnAppsCreateContextMenu, selectedListItem: AppsListItem) {
+        viewHolder.itemView.setOnCreateContextMenuListener { menu, v, _ ->
+            onAppsCreateContextMenu.onAppsCreateContextMenu(menu, v, selectedListItem)
+        }
     }
 
     override fun getItemId(position: Int): Long {
         return position.toLong()
     }
 
-    override fun getCount(): Int {
+    override fun getItemCount(): Int {
         return appsAndSeparators.size
     }
 
-    override fun getViewTypeCount(): Int {
-        return ITEM_VIEW_TYPE_COUNT
-    }
-
     override fun getItemViewType(position: Int): Int {
         return when (appsAndSeparators[position]) {
             is AppItem -> ITEM_VIEW_TYPE_APP
             is AppSeparatorItem -> ITEM_VIEW_TYPE_SEPARATOR
         }
     }
 
-    override fun isEnabled(position: Int): Boolean {
-        return when (appsAndSeparators[position]) {
-            is AppItem -> true
-            is AppSeparatorItem -> false
-        }
+    override fun onViewDetachedFromWindow(holder: ViewHolder) {
+        holder.itemView.clearAnimation()
+    }
+
+    fun getLastSelectedAppsListItem(): AppsListItem {
```

Comment:
Can you rename these functions to indicate that their in relation to context items? Otherwise it seems too much like something that should be handled by the viewmodel.

Reply:
How does `lastSelectedAppContextItem` sound?

Reply:
Perfect